### PR TITLE
fix: overflow layout memory leaks

### LINKED ISSUE #9345: overflow layout memory leaks
#### Is this a bug, enhancement, or feature request?
bug
#### Briefly describe your proposal.
Currently, overflow layout does not completely destroys, thus keeping links to elements/components/directives inside.

How to reproduce:
Follow this instructions on how to collect detached elements:
https://learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/memory-problems/#discover-detached-dom-tree-memory-leaks-with-heap-snapshots

Go to overflow layout page, then go to home page, run Collect garbage and then click "Get Detached elements"

### PR DESCRIPTION
**Pull request by N1XUS**
## Related Issue(s)

closes #9345

## Description
- Fixed overflow layout memory leaks;
- Fixed memory leaks from focusKeyManager if used with QueryList;
- Fixed animations in dialog;
- Unit tests are now using NoopAnimations module instead of BrowserAnimationsModule;
- Fixed dialog and message box memory leaks;

### CODE REVIEW

**Review by g-cheishvili — APPROVED** [review]

File: libs/core/src/lib/overflow-layout/directives/overflow-layout-item.directive.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export class OverflowLayoutItemDirective implements OverflowItem, OnInit {
 
     /** @hidden */
     ngOnInit(): void {
-        this._overflowItemRef?.setElementRef(this.elmRef);
+        // this._overflowItemRef?.setElementRef(this.elmRef);
```
Comment (platon-rov) [inline libs/core/src/lib/overflow-layout/directives/overflow-layout-item.directive.ts:70]: can be removed?

File: libs/core/src/lib/overflow-layout/overflow-layout.component.ts [hunk 2]
```diff
@@ -309,17 +326,46 @@ export class OverflowLayoutComponent implements OnInit, AfterViewInit, OnDestroy
...
-                        this._overflowLayoutService._keyboardEventsManager.onKeydown(event);
+                        this._keyboardEventsManager?.onKeydown(event);
                     }
                 })
         );
     }
+
+    /** @hidden */
+    private _setFocusKeyManager(): void {
+        if (!this.enableKeyboardNavigation) {
```
Comment (platon-rov) [inline libs/core/src/lib/overflow-layout/overflow-layout.component.ts:345]: can this be done with focusable-list cdk thing? probably not in scope of this pr

File: wdio_docs-core-button.conf.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports.config = require('./e2e/wdio.base-config.js')({
```
Comment (platon-rov) [inline wdio_docs-core-button.conf.js:1]: probably should be removed.

File: libs/core/src/lib/overflow-layout/overflow-layout.component.ts (hunk 2 above)
Comment (N1XUS) [reply libs/core/src/lib/overflow-layout/overflow-layout.component.ts:345]: I would suggest other PR for that

**Pull request merged**